pkg/kubelet/certificate/certificate_manager.go

```diff
@@ -160,9 +148,19 @@ func (m *manager) shouldRotate() bool {
 	m.certAccessLock.RLock()
 	defer m.certAccessLock.RUnlock()
 	notAfter := m.cert.Leaf.NotAfter
-	total := notAfter.Sub(m.cert.Leaf.NotBefore)
-	remaining := notAfter.Sub(time.Now())
-	return remaining < 0 || uint(remaining*100/total) < m.shouldRotatePercent
+	totalDuration := float64(notAfter.Sub(m.cert.Leaf.NotBefore))
+
+	// Use some jitter to set the rotation threshold so each node will rotate
+	// at approximately 70-90% of the total lifetime of the certificate.  With
+	// jitter, if a number of nodes are added to a cluster at approximately the
+	// same time (such as cluster creation time), they won't all try to rotate
+	// certificates at the same time for the rest of the life of the cluster.
+	//
+	// For the parameters of the Jitter function:
+	//    [total*0.7, total+total*0.7*0.28] ~ [total*0.7, total*0.9]
+	jitteryDuration := wait.Jitter(time.Duration(totalDuration*0.7), 0.28)
```

Comment:
I'm lost in this.
If I understand correctly, what you would like to achieve is that:
- certificate created at t1, expires at t2 (dt := t2 - t1)
- you would like it to be refreshed not earlier than t1 + 0.7dt, and not later than t1 + 0.9dt
Is that right?

If so, this should be:
```
jitteryDuration := wait.Jitter(time.Duration(totalDuration*0.2), 1.0) + time.Duration(totalDuration*0.7)
```